test/regtest.py

```diff
@@ -69,81 +81,158 @@ def metadata(file):
         m['expect'] = match.group(1).strip()
 
   if not m['skip'] and not 'expect' in m:
-    print red("WARNING: @expect MISSING IN %s" % file)
+    logging.warning("@expect MISSING IN %s" % file)
     m['expect'] = 'verified'
 
   return m
-  
-parser = argparse.ArgumentParser()
-parser.add_argument("--exhaustive", help="be exhaustive", action="store_true")
-args = parser.parse_args()
 
-print "Running regression tests..."
-print
+# integer constants
+PASSED = 0; TIMEDOUT = 1; UNKNOWN = 2; FAILED = -1;
+def process_test(cmd, test, memory, verifier, expect, log_file):
+    """
+    This is the worker function for each process. This function process the supplied
+    test and returns a tuple containing  indicating the test results.
+
+    :return: A tuple with the
+    """
+    str_result = "{0:>20}\n".format(test)
+    str_result += "{0:>20} {1:>10}    :".format(memory, verifier)
+
+    t0 = time.time()
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err  = p.communicate()
+    elapsed = time.time() - t0
+
+
+
+    # get the test results
+    result = get_result(out+err)
+    if result == expect:
+      str_result += 'PASSED '
+      str_result += '  [%.2fs]' % round(elapsed, 2)
+      return green(str_result, log_file)
```

Comment:
Only the string `'PASSED'` was green — see how the existing regression script works.